crates/data_components/src/flight.rs

```diff
@@ -538,32 +538,39 @@ fn query_to_stream(
 fn to_execution_error(e: Error) -> DataFusionError {
     match e {
         Error::Flight { source } => match source {
-            flight_client::Error::UnableToQuery { source } => {
-                DataFusionError::Execution(format!("{source}"))
+            flight_client::Error::UnableToQuery {
+                source: source_error,
+            } => DataFusionError::Execution(format!("{source_error}")),
+            flight_client::Error::ConnectionReset { source: _ } => {
+                DataFusionError::External(Box::new(source))
             }
             _ => DataFusionError::Execution(format!("{source}")),
         },
         _ => DataFusionError::Execution(format!("{e}")),
     }
 }
 
-fn map_connection_reset_error(error: FlightError) -> Error {
+fn map_query_stream_error(error: FlightError) -> DataFusionError {
     match error {
         FlightError::Tonic(ref source) => {
             let source_owned = *source.clone();
             if is_connection_reset_error(&source_owned) {
-                return Error::ArrowFlight {
-                    source: Box::new(FlightClientError::ConnectionReset {
-                        source: TonicStatusError::from(source_owned),
-                    }),
-                };
+                return DataFusionError::External(Box::new(FlightClientError::ConnectionReset {
+                    source: TonicStatusError::from(source_owned),
+                }));
             }
+            DataFusionError::Execution(format!(
+                "{}",
+                Error::ArrowFlight {
+                    source: Box::new(error),
+                }
+            ))
```

Comment:
```suggestion
            DataFusionError::Execution(
                Error::ArrowFlight {
                    source: Box::new(error),
                }
                .to_string(),
            )
```

nit: There is no need to use `format!()` when the error already implements `Display`.